redo/__init__.py

```diff
@@ -56,21 +56,32 @@ def retrier(attempts=5, sleeptime=10, max_sleeptime=300, sleepscale=1.5, jitter=
         ...     print "max tries hit"
         max tries hit
     """
+    if jitter > sleeptime:
+        # To prevent negative sleep times
+        raise Exception('jitter (%i) must be less than sleep time (%i)', jitter, sleeptime)
+
+    sleeptime_real = sleeptime
     for _ in range(attempts):
         log.debug("attempt %i/%i", _ + 1, attempts)
-        yield
+
+        yield sleeptime_real
+
         if jitter:
-            sleeptime += random.randint(-jitter, jitter)
-            sleeptime = max(sleeptime, 0)
-
-        if _ == attempts - 1:
-            # Don't need to sleep the last time
-            break
-        log.debug("sleeping for %.2fs (attempt %i/%i)", sleeptime, _ + 1, attempts)
-        time.sleep(sleeptime)
+            sleeptime_real = sleeptime + random.randint(-jitter, jitter)
+            # our jitter should scale along with the sleeptime
+            jitter = int(jitter * sleepscale)
+        else:
+            sleeptime_real = sleeptime
+
         sleeptime *= sleepscale
```

Comment:
I think you've broken sleepscale for non-jitter cases here. I _think_ this line needs a s/sleeptime/sleeptime_real/. Looks like this has uncovered a gap in the test cases too.